ext/lmdb-safe/lmdb-safe.hh

```diff
@@ -80,6 +80,7 @@ public:
   int getROTX();
   void incROTX();
   void decROTX();
+  bool d_lmdb_safe_IDs_are_random{false};
```

Comment:
It feels a bit weird to store that inside `MDBEnv`, perhaps we could store it inside `LMDBBackend` and pass it to `RWTransaction::put()` instead?